spec/system/admin/reports/enterprise_summary_fees/enterprise_summary_fee_with_tax_report_by_producer_spec.rb

```diff
@@ -31,17 +31,20 @@
   }
   let!(:ship_address){ create(:ship_address) }
 
-  let!(:variant){ create(:variant, tax_category: tax_category) }
-  let!(:product){ variant.product }
-  let!(:distributor){ create(:distributor_enterprise_with_tax, name: 'Distributor') }
   let!(:supplier){ create(:supplier_enterprise, name: 'Supplier', charges_sales_tax: true) }
+  let!(:supplier2){ create(:supplier_enterprise, name: 'Supplier2', charges_sales_tax: true) }
+  let!(:product){ create(:simple_product, supplier: supplier ) }
+  let!(:product2){ create(:simple_product, supplier: supplier2 ) }
+  let!(:variant){ create(:variant, product_id: product.id, tax_category: tax_category) }
+  let!(:variant2){ create(:variant, product_id: product2.id, tax_category: tax_category) }
+  let!(:distributor){ create(:distributor_enterprise_with_tax, name: 'Distributor') }
   let!(:payment_method){ create(:payment_method, :flat_rate) }
   let!(:shipping_method){ create(:shipping_method, :flat_rate) }
 
   let!(:order){ create(:order_with_distributor, distributor: distributor) }
   let!(:order_cycle){
-    create(:simple_order_cycle, suppliers: [supplier], distributors: [distributor],
-                                variants: [variant], name: "oc1")
+    create(:simple_order_cycle, suppliers: [supplier, supplier2], distributors: [distributor],
+                                variants: [variant, variant2], name: "oc1")
```

Comment:
I finally found out the issue: adding the variants to the order cycle in this manner created exchanges in such a way, that enterprise fees were being applied to both incoming exchanges, regardless of the supplier. 

This was quite intriguing, as I could not spot any difference in the DB... Anyway, it's not a bug, and a set-up is proposed in the next commit (to be squashed with this one).